### PR TITLE
feat : update DTO package  (create and update) , update response message

### PR DESCRIPTION
**Pull request by darship19**

### CODE REVIEW

File: inventory-service/src/main/java/com/inventoryservice/Configs/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package com.inventoryservice.Configs;
+
+public class GlobalExceptionHandler {
+}
```
Comment (SathurshanPrabaharan) [inline inventory-service/src/main/java/com/inventoryservice/Configs/GlobalExceptionHandler.java:4]: You need to implement the this exception handler

File: inventory-service/src/main/java/com/inventoryservice/DTO/Request/InventoryUpdateRequest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    private String shortNote;
+    @PastOrPresent(message = "Invalid Last Maintenance Date: Last Maintenance Date cannot be in the future")
+    private LocalDate lastMaintenanceDate;
+    @Future(message = "Invalid Next Maintenance Date: Next Maintenance Date must be in the future")
+    private LocalDate nextMaintenanceDate;
+    private LocalDateTime createdAt;
+    private LocalDateTime updatedAt;
+    @NotNull(message = "Invalid CreatedBy: CreatedBy cannot be null")
+    private Long createdBy;
+    private List<Integer> installedSoftwares;
```
Comment (SathurshanPrabaharan) [inline inventory-service/src/main/java/com/inventoryservice/DTO/Request/InventoryUpdateRequest.java:64]: change this as installed software Ids
then find these software and add to related inventory

File: inventory-service/src/main/java/com/inventoryservice/DTO/Response/Inventory/InventoryDetailsResponse.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    private LocalDate purchaseDate;
+    private Float purchaseCost;
+    private LocalDate warrantyExpiry;
+    private String shortNote;
+    private LocalDate lastMaintenanceDate;
+    private LocalDate nextMaintenanceDate;
+    private LocalDateTime createdAt;
+    private LocalDateTime updatedAt;
+    private Long createdBy;
+    private List<Integer> installedSoftwares;
```
Comment (SathurshanPrabaharan) [inline inventory-service/src/main/java/com/inventoryservice/DTO/Response/Inventory/InventoryDetailsResponse.java:54]: why List<Integer> ??
modify the software then use it
like List<ModifiedSoftware>

File: inventory-service/src/main/java/com/inventoryservice/Services/Impls/InventoryServiceImpl.java [hunk 4]
```diff
@@ -1,145 +1,97 @@
...
 
-    public Inventory getInventory(UUID id) throws InventoryNotFoundException {
-        Inventory inventory = inventoryRepository.findById(id)
-                .orElseThrow(()->new InventoryNotFoundException("Inventory not found with ID: " + id));
-        return inventory;
 
+    @Override
+    public Inventory findById(UUID id) {
+        Optional<Inventory> AdminOptional = inventoryRepository.findById(id);
+        return AdminOptional.orElseThrow(() -> new InventoryNotFoundException("admin not found with id : " + id));
```
Comment (SathurshanPrabaharan) [inline inventory-service/src/main/java/com/inventoryservice/Services/Impls/InventoryServiceImpl.java:55]: why AdminOptional ???
change the exception message

File: inventory-service/src/main/java/com/inventoryservice/Models/Inventory.java [hunk 5]
```diff
@@ -19,6 +20,7 @@
 @Data
 @NoArgsConstructor
 @AllArgsConstructor(staticName = "build")
+@Builder
 public class Inventory {
 
     @Id
```
Comment (SathurshanPrabaharan) [inline inventory-service/src/main/java/com/inventoryservice/Models/Inventory.java:26]: @Column(name = "installed_softwares")
    private List<Integer> installedSoftwares;
    map using many to many relationship.
    better first implement the software then do this

**Review by SathurshanPrabaharan — CHANGES_REQUESTED** [review]

File: inventory-service/src/main/java/com/inventoryservice/Models/Inventory.java (hunk 5 above)
Comment (darship19) [reply inventory-service/src/main/java/com/inventoryservice/Models/Inventory.java:26]: @ManyToMany
    @JoinTable(
            name = "inventory_software",
            joinColumns = @JoinColumn(name = "inventory_id"),
            inverseJoinColumns = @JoinColumn(name = "software_id")
    )
    private List<Software> installedSoftwares;

**Pull request closed without merging**